### PR TITLE
Documentation related to listing user's actions on application

### LINKED ISSUE #474: Documentation on actions displaying on each application for different users
This comment on [PR#227BE](https://github.com/openmsupply/application-manager-server/pull/266#discussion_r598024167) explain how we do it:
> Start -> review_assigned_not_started_count
Re-Review-> review_pending_count
Update -> review_change_request_count
Continue -> review_draft_count
Self-Assign -> review_available_for_self_assignment_count

>review_assigned_count -> not really sure what to indicate as action, but the idea is it can be filtered by it

>Re-Review and Update would change to 'Continue' in application list (but would be correctly represented in review_home

Some little more explanation and examples will be very good to have in somewhere for quick access.

### PR DESCRIPTION
**Pull request by nmadruga**
Fixes #474 

### based on PR
#623 

### Changes
- Add actions [Status cell document](https://app.diagrams.net/#G1IGiHyFRdMIMW3xUBNCw5GMz9GUdSKtpC) flow to be published in the wiki
- list possible actions per role and basic definition of when each is available (and how they are generated based on application_list_view fields)

### CODE REVIEW

File: documentation/internal/Actions-on-Applications.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+Actions are available to the applicant user for their own applications, and very basic for now:
+
+- `Continue` -> Application created that hasn't been submitted
+- `View` -> Application that has been submitted or after application process is completed
+- `Update` -> After an application has been submitted and reveiwed, when there are **changes required** to the Applicant
+
+### Other applicant's actions (todo)
+
+- `Widthdrawn` -> Application that has been submitted
```
Comment (CarlosNZ) [inline documentation/internal/Actions-on-Applications.md:23]:
```suggestion
- `Withdraw` -> Application that has been submitted
```

File: documentation/internal/Actions-on-Applications.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+- `Delete` -> Application that hasn't been submitted
+
+## Reveiwer's or Consolidator's actions
```
Comment (CarlosNZ) [inline documentation/internal/Actions-on-Applications.md:26]:
```suggestion
## Reviewer's or Consolidator's actions
```

File: documentation/internal/Actions-on-Applications.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+Actions are available to the Reviewer user (based `template_permission`) after the **application is submitted**.
```
Comment (CarlosNZ) [inline documentation/internal/Actions-on-Applications.md:28]:
```suggestion
Actions are available to the Reviewer user (based on `template_permission`) after the **application is submitted**.
```

File: documentation/internal/Actions-on-Applications.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+The options are per section of the application, but will show on the application list as the top most possible one (combining the actions available for all sections in the application):
+
+- `Assign` -> When no reviewer is assigned yet (checks `review_question_assignment`)
+- `Re-assign` -> When reviewer is assigned
+
+**Note**: When re-assigning section(s) when the review has been started there is some pending work todo for moving assignments
+
+### Application list view
+
+On server side we use smart view to calculate the count of each of the following (used to define available actions on the front-end) based on status of `review_assignment` and `review` (associated to self for Reviewer, and associated to any review for Assigners).
```
Comment (CarlosNZ) [inline documentation/internal/Actions-on-Applications.md:49]: Could you perhaps specify what you mean by "associated to"? You mean using the "user" as the parameter for the back-end Function, right? Maybe just explain that a little more?

**Review by CarlosNZ — CHANGES_REQUESTED** [review]
Great, thanks for explaining all this, Nicole. It's very helpful.

Just a couple of minor typos etc. Cheers.

File: documentation/internal/Actions-on-Applications.md (hunk 4 above)
Comment (nmadruga) [reply documentation/internal/Actions-on-Applications.md:49]: I can rephrase this better : ) will do tomorrow morning though

Comment (nmadruga) [reply documentation/internal/Actions-on-Applications.md]: @CarlosNZ I tried to rephrase this, and actually realise it wasn't accurate, so hopefully now it's better to understand and makes sense: 2a1079f

File: documentation/internal/Actions-on-Applications.md [hunk 5]
```diff
@@ -57,3 +58,5 @@ The actions are available to the user (if permissions apply) when the related fi
 - `View` -> review_submitted_count
 - `Assign` -> assign_count
 - `Re-assign` -> assign_count & is_fully_assigned_level_1
+
+**Note**: For now only the front-end is displaying all reviews associated to the current application on the "Review Home page", later when restrictions apply only users with higher permissions will be able to see other users review's status.
```
Comment (CarlosNZ) [inline documentation/internal/Actions-on-Applications.md:62]:
```suggestion
**Note**: For now the front-end is displaying all reviews associated with the current application on the "Review Home page". Later, when restrictions apply, only users with appropriate permissions will be able to see other users' review statuses.
```

File: documentation/internal/Actions-on-Applications.md [hunk 6]
```diff
@@ -46,7 +46,8 @@ The options are per section of the application, but will show on the application
 
 ### Application list view
 
-On server side we use smart view to calculate the count of each of the following (used to define available actions on the front-end) based on status of `review_assignment` and `review` (associated to self for Reviewer, and associated to any review for Assigners).
+On server side we use smart view to calculate the count of each of the following (used to define available actions on the front-end) based on status of `review_assignment` and `review`.
+The GraphQL query to application list view will fetch ALL `review_assignment` related to the current application `serial`, for all reviewer `user` of any `level` and `stage`.
```
Comment (CarlosNZ) [inline documentation/internal/Actions-on-Applications.md:50]: The GraphQL query to application list view -- might be good to include the name of the query (in brackets) here.

**Review by CarlosNZ — CHANGES_REQUESTED** [review]
Cool, if you could just make the clarifications we discussed.

Cheers.

File: documentation/internal/Actions-on-Applications.md [hunk 7]
```diff
@@ -44,19 +44,41 @@ The options are per section of the application, but will show on the application
 
 **Note**: When re-assigning section(s) when the review has been started there is some pending work todo for moving assignments
 
-### Application list view
+## Generating and display of actions
 
-On server side we use smart view to calculate the count of each of the following (used to define available actions on the front-end) based on status of `review_assignment` and `review`.
-The GraphQL query to application list view will fetch ALL `review_assignment` related to the current application `serial`, for all reviewer `user` of any `level` and `stage`.
-The actions are available to the user (if permissions apply) when the related field(s) is above 0 or `true`.
+Actions are displayed in 2 main places: on the Application's list and on the Review home page (for Reviewer/Assigner) or Application home page (for Applicant)
```
Comment (CarlosNZ) [inline documentation/internal/Actions-on-Applications.md:49]:
```suggestion
Actions are displayed in 2 main places: on the Applications list and on the Review home page (for Reviewer/Assigner) or Application home page (for Applicant)
```

File: documentation/internal/Actions-on-Applications.md [hunk 8]
```diff
@@ -44,19 +44,41 @@ The options are per section of the application, but will show on the application
...
+
+**Note**: For now the front-end is displaying all reviews associated with the current application on the "Review Home page". Later, when restrictions apply, only users with appropriate permissions will be able to see other users' review statuses.
+
+Actions generated for **assigner** user will check which of the count (of ALL review/assignment) field is above 0 or `true`,
+
 - `Assign` -> assign_count
 - `Re-assign` -> assign_count & is_fully_assigned_level_1
 
-**Note**: For now the front-end is displaying all reviews associated with the current application on the "Review Home page". Later, when restrictions apply, only users with appropriate permissions will be able to see other users' review statuses.
+**Note**: If user is both an Assigner and Review for the current application the actions to both can disoplay to them, one for Reviewer and one for Assigner.
```
Comment (CarlosNZ) [inline documentation/internal/Actions-on-Applications.md:84]:
```suggestion
**Note**: If user is both an Assigner and Reviewer for the current application the actions for _both_ will be displayed — one for Reviewer and one for Assigner.
```

File: documentation/internal/Actions-on-Applications.md [hunk 9]
```diff
@@ -44,19 +44,41 @@ The options are per section of the application, but will show on the application
...
+
+### Applicant home page (or list)
+
+The GraphQL query (`getApplication` and `getApplicationList`) simply gets application related to the current user and the `status`, which is used to define the action to be displayed.
+
+- `Continue` -> Application with `status` = DRAFT (and owned by the current user)
+- `View` -> Application with `status` = SUBMITTED or others (e.g. COMPLETED, EXPIRED, ...)
+- `Update` -> Application with `status` = CHANGES_REQUIRED
+
+**Note**: For now the front-end allows to see any application, not only restricted by user permission. Later, when restrictions apply, only users with appropriate permissions will be able to see other users' application statuses.
```
Comment (CarlosNZ) [inline documentation/internal/Actions-on-Applications.md:59]:
```suggestion
**Note**: For now the front-end allows a user to see any application, not only restricted by user permission. Later, when restrictions apply, only users with appropriate permissions will be able to see other users' application statuses.
```

**Review by CarlosNZ — APPROVED** [review]
Great, very helpful.

Just a couple of minor typos, then it's good to go :)

**Comment by nmadruga** [comment]
Will be waiting for PR #623 to be merged before this one can be merged into master too

**Pull request merged**